great_expectations/data_context/data_context.py

```diff
@@ -2432,13 +2439,17 @@ def validations_store_name(self):
     def validations_store(self):
         return self.stores[self.validations_store_name]
 
-    def _compile_evaluation_parameter_dependencies(self):
+    def _compile_evaluation_parameter_dependencies(self, expectation_suite):
         self._evaluation_parameter_dependencies = {}
-        for key in self.expectations_store.list_keys():
-            expectation_suite = self.expectations_store.get(key)
-            if not expectation_suite:
-                continue
 
+        # only if we don't have an expectation suite do we do a key scan
+        if(expectation_suite is None):
```

Comment:
Why use this pattern instead of `if expectation_suite is None`?

Reply:
fixed